pkgs/applications/audio/gst123/default.nix

```
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchurl, pkg-config, wrapGAppsHook, gst_all_1, gtk3, ncurses }:
+
+stdenv.mkDerivation rec {
+  pname = "gst123";
+  version = "0.4.0";
+  src = fetchurl {
+    url = "https://github.com/swesterfeld/gst123/releases/download/${version}/gst123-${version}.tar.bz2";
+    sha256 = "sha256-ThE7Us1ZGNjrsG8DkdL3NXk9s6bcW6F3+krghRnJzAQ=";
+  };
```

Use this:


```suggestion
  src = fetchFromGitHub {
    owner = "swesterfeld";
    repo = "gst123";
    rev = version;
    hash = "";
  };
```

(update the hash.)

---

That would break the build because gst123-0.4.0 assumes that it can access .git if it was build from git checkout. Nevertheless I looked at the code that needs .git, and its really legacy code for ChangeLog (which I think is no longer needed since developers usually have history via git), so since I'm upstream developer I'll go with your suggestion and make a gst123-0.4.1 release that doesn't need .git when building from git.

---

Usually it is not a good idea to rely on `.git`, since it is not a really stable interface.
Also, for us on Nixpkgs, it introduces too much impurity.

There are some flags we pass for those rare cases when `.git` is really needed, tho.